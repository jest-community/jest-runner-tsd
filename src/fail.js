const { toTestResult } = require('./toTestResult');

/**
 * @typedef {object} Test
 * @property {string} path
 * @property {string} title
 */

/**
 * @typedef {object} Options
 * @property {number} start
 * @property {number} end
 * @property {Test} test
 * @property {number} numFailed
 * @property {number} numPassed
 * @property {string=} errorMessage
 */

/**
 * @param {Options} options
 */
module.exports.fail = ({
  start,
  end,
  test,
  errorMessage,
  numFailed,
  numPassed,
}) => {
  const stats = {
    failures: numFailed,
    passes: numPassed,
    pending: 0,
    todo: 0,
    end,
    start,
  };

  return toTestResult({
    stats,
    errorMessage,
    skipped: false,
    tests: [{ duration: end - start, ...test, errorMessage }],
    jestTestPath: test.path,
  });
};
