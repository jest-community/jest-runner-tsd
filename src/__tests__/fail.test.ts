import { expect, test } from '@jest/globals';
import { fail } from '../fail';

const errorMessage = `  ● tsd typecheck

  Argument of type 'number' is not assignable to parameter of type 'string'.

  > 5 | expectType<string>(concat(1, 2));
      | ^

    at e2e/__fixtures__/js-failing/index.test.ts:5:1`;

test('fails', () => {
  const testResult = fail({
    start: 0,
    end: 1200,
    test: {
      path: '/path/to/e2e/__fixtures__/js-failing/index.test.ts',
      title: 'tsd typecheck',
    },
    errorMessage,
    numFailed: 1,
    numPassed: 3,
  });

  expect(testResult).toMatchSnapshot();
});
