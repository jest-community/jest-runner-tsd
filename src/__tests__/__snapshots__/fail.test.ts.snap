// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fails 1`] = `
Object {
  "failureMessage": "  ● tsd typecheck

  Argument of type 'number' is not assignable to parameter of type 'string'.

  > 5 | expectType<string>(concat(1, 2));
      | ^

    at e2e/__fixtures__/js-failing/index.test.ts:5:1",
  "leaks": false,
  "numFailingTests": 1,
  "numPassingTests": 3,
  "numPendingTests": 0,
  "numTodoTests": 0,
  "openHandles": Array [],
  "perfStats": Object {
    "end": 1200,
    "runtime": 1200,
    "slow": false,
    "start": 0,
  },
  "skipped": false,
  "snapshot": Object {
    "added": 0,
    "fileDeleted": false,
    "matched": 0,
    "unchecked": 0,
    "uncheckedKeys": Array [],
    "unmatched": 0,
    "updated": 0,
  },
  "testFilePath": "/path/to/e2e/__fixtures__/js-failing/index.test.ts",
  "testResults": Array [
    Object {
      "ancestorTitles": Array [],
      "duration": 1200,
      "failureDetails": Array [],
      "failureMessages": Array [
        "  ● tsd typecheck

  Argument of type 'number' is not assignable to parameter of type 'string'.

  > 5 | expectType<string>(concat(1, 2));
      | ^

    at e2e/__fixtures__/js-failing/index.test.ts:5:1",
      ],
      "fullName": "/path/to/e2e/__fixtures__/js-failing/index.test.ts",
      "numPassingAsserts": 1,
      "status": "failed",
      "title": "tsd typecheck",
    },
  ],
}
`;
